### PR TITLE
Enable using stdin when starting docker compose

### LINKED ISSUE #285: Feature request: Enable using stdin when starting docker compose
It would be a nice feature if one can use stdin to inject a docker compose file on the fly instead of using a file path. Docker Compose supports this. Wenn giving ``-`` as filename to command line option ``-f`` docker compose reads the configuration from stdin.

Maybe somewhere [here](https://github.com/testcontainers/testcontainers-go/blob/master/compose.go#L160) this can be done (but I am not good enough to make an acceptable pull request).

### PR DESCRIPTION
**Pull request by mal-as**
Closes #285

### CODE REVIEW

File: compose.go [hunk 1]
```diff
@@ -402,3 +416,29 @@ func which(binary string) error {
...
+	return ioutil.ReadAll(os.Stdin)
+}
+
+func (dc *LocalDockerCompose) makeTmpFile() error {
+	data, err := readSTDIN()
+	if err != nil {
+		return err
+	}
+
+	file, err := ioutil.TempFile("/Users/aleksamalyshev/", "__tmp_")
```
Comment (mdelapenya) [inline compose.go:430]: I think this is a leftover 😅

File: compose.go [hunk 2]
```diff
@@ -402,3 +416,29 @@ func which(binary string) error {
 
 	return err
 }
+
+func readSTDIN() ([]byte, error) {
+	return ioutil.ReadAll(os.Stdin)
+}
+
+func (dc *LocalDockerCompose) makeTmpFile() error {
+	data, err := readSTDIN()
```
Comment (mdelapenya) [inline compose.go:425]: Better in one line?
```suggestion
func (dc *LocalDockerCompose) makeTmpFile() error {
	data, err := ioutil.ReadAll(os.Stdin)
```

File: compose.go [hunk 3]
```diff
@@ -402,3 +416,29 @@ func which(binary string) error {
...
+	if err != nil {
+		return err
+	}
+
+	file, err := ioutil.TempFile("", "__tmp_")
```
Comment (mdelapenya) [inline compose.go:430]: Do you foresee any collision between multiple runs? Do you think we could use the compose identifier as part of the file name in that case 🤔?

File: compose.go [hunk 4]
```diff
@@ -103,8 +104,16 @@ func NewLocalDockerCompose(filePaths []string, identifier string, opts ...LocalD
 
 	dc.absComposeFilePaths = make([]string, len(filePaths))
 	for i, cfp := range dc.ComposeFilePaths {
-		abs, _ := filepath.Abs(cfp)
-		dc.absComposeFilePaths[i] = abs
+		if cfp == "-" {
```
Comment (mdelapenya) [inline compose.go:107]: Mmm I do not get this line. Is it expected to receive a `-` as a file path for a compose file?

**Review by mdelapenya — COMMENTED** [review]
Thanks for your work with this PR. I left a few review comments before merging it. Please let me know what you think

**Comment by mdelapenya** [comment]
@mal-as  #470 was created yesterday, and I think it collides with this one. What are your thoughts on that PR?

File: compose.go (hunk 4 above)
Comment (mal-as) [reply compose.go:107]: Yes, it is

File: compose.go (hunk 3 above)
Comment (mal-as) [reply compose.go]: fixed it

**Comment by mdelapenya** [comment]
I think this PR is superseded by #466, therefore I'm closing it. @mal-as could you please reopen it if you feel it's different? Thanks!

**Pull request closed without merging**